codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -224,6 +280,10 @@ private void decodeParams(String s) {
                     name = decodeComponent(s.substring(pos, i), charset);
                 }
                 pos = i + 1;
+            } else if (withCommaDecoding && c == ',' && name != null) {
```

Comment:
I'm not overly familiar with the RFC but this was my assumption as well. It seems like there are conflicting thought processes here:

1. https://github.com/netty/netty/pull/5820#r78891898 - `I'd tend to stick close to application/x-www-form-urlencoded spec and only interpret comma as a delimiter inside a value. So here we'd have ((",", "f")).`
2. https://github.com/netty/netty/pull/5820#r79009061 - `If we find a sub-delimiter in an URI string, it can only mean it's a sub-delimiter for sure, otherwise it should have been encoded.`

Where I'm confused is if we follow 2 and treat `,` as a delimiter always ... then it seems like 1 is only considering part of the picture. Based on the definition of `query` shouldn't we follow 2 and always treat `,` as a delimiter?

> pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
> query       = *( pchar / "/" / "?" )